File: api/src/main/java/com/pk/ysf/service/mapper/BaseMapper.java
package com.pk.ysf.service.mapper;

import java.util.Collection;
import java.util.stream.Collectors;

@FunctionalInterface
public interface BaseMapper<F, T> {

    T map(F object);

    default Collection<T> mapAll(Collection<F> collections) {
        return collections.stream()
                .map(this::map)
                .collect(Collectors.toList());
    }
}

File: api/src/main/kotlin/com/pk/ysf/service/mapper/BaseMapper.kt
package com.pk.ysf.service.mapper

import java.util.stream.Collectors

interface BaseMapper<F, T> {

    fun map(from: F): T

    fun mapAll(collections: Collection<F>): Collection<T> {
        return collections.stream()
                .map { from -> map(from) }
                .collect(Collectors.toList())
    }
}